.changeset/afraid-numbers-invite.md

```diff
@@ -0,0 +1,15 @@
+---
+'@backstage/repo-tools': minor
+---
+
+Updates the OpenAPI client template to support the new format for identifying plugin ID. You should now use `info.title` like so,
+
+```diff
+info:
++    title: yourPluginId
+-    title: @internal/plugin-*-backend
```

Comment:
An alternative approach might be to use the [`--additional-properties=`](https://openapi-generator.tech/docs/configuration) options for the CLI and either autopopulate that from the package context or pull it in from the command line. I think I like that better, more intuitive as a user to just create a client from a given backend. Will rework this.

Reply:
There's not a great source of `pluginId` from the package right now though. It's something we'll add in the future for sure though. Let's not derive it from the package name or file structure :p

Will leave it to you whether you want to merge this and tweak it in the future or go for a different approach straight away. I think it could make sense to ship tbh, as a single better state than the current iteration as adoption of this ramps up.

Reply:
True, let's iterate on this.